core/templates/domain/topics_and_skills_dashboard/topics-and-skills-dashboard-backend-api.service.ts

```diff
@@ -17,27 +17,34 @@
   from the backend and to merge skills from the dashboard.
  */
 
-require(
-  'domain/topics_and_skills_dashboard/' +
-  'topics-and-skills-dashboard-domain.constants.ajs.ts');
+import { downgradeInjectable } from '@angular/upgrade/static';
+import { Injectable } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+import { TopicsAndSkillsDashboardDomainConstants } from
+  // eslint-disable-next-line max-len
+  '../topics_and_skills_dashboard/topics-and-skills-dashboard-domain.constants';
 
-angular.module('oppia').factory('TopicsAndSkillsDashboardBackendApiService', [
-  '$http', 'MERGE_SKILLS_URL', function($http, MERGE_SKILLS_URL) {
-    var _fetchDashboardData = function() {
-      return $http.get('/topics_and_skills_dashboard/data');
-    };
 
-    var _mergeSkills = function(oldSkillId, newSkillId) {
-      var mergeSkillsData = {
-        old_skill_id: oldSkillId,
-        new_skill_id: newSkillId
-      };
-      return $http.post(MERGE_SKILLS_URL, mergeSkillsData);
-    };
+@Injectable({
+  providedIn: 'root'
+})
+export class TopicsAndSkillsDashboardBackendApiService {
+  constructor(private http: HttpClient) {}
+
+  fetchDashboardData(): Promise<any> {
+    return this.http.get('/topics_and_skills_dashboard/data').toPromise();
+  }
 
-    return {
-      fetchDashboardData: _fetchDashboardData,
-      mergeSkills: _mergeSkills
+  mergeSkills(oldSkillId, newSkillId): Promise<void> {
```

Comment:
Should we have types for "oldSkillId" and "newSkillId"?

Reply:
Also, can you explain why we use `Promise<void>` here?

Reply:
oldSkillId and newSkillId should be string. I was getting issues earlier in the post query then asked one of the mentors, he suggested me to changes this (and the post<void>).

It also works fine when i use datatypes  

mergeSKills(oldSKillId:string, newSkillId:string) : Promise<Object> and
this.http.post ()  --removing void here

Reply:
I suggested `Promise<void>` to be explicit about the return type (`Promise<void>` means you can only wait on it, nothing will be returned), although dunno if there's a style guide suggestion for this.

@bansalnitish WDYT?

Reply:
Ah, makes sense. I'm fine with that. Do you think we should mention this in the migration instructions page [here](https://github.com/oppia/oppia/wiki/Angular-Migration)?